source/client/options_impl.cc

```diff
@@ -482,39 +537,67 @@ void OptionsImpl::validate() const {
       throw MalformedArgvException("Value for --concurrency should be greater then 0.");
     }
   }
-  try {
-    UriImpl uri(uri_);
-  } catch (const UriException&) {
-    throw MalformedArgvException("Invalid URI");
+  if (uri_.has_value()) {
+    try {
+      UriImpl uri(uri_.value());
+    } catch (const UriException&) {
+      throw MalformedArgvException("Invalid URI");
```

Comment:
(pre-existing, optional) Can we include the invalid uri in the error message?

Reply:
Done